nilearn/regions/tests/test_region_extractor.py

```diff
@@ -296,17 +329,11 @@ def test_remove_small_regions():
     assert sum_removed_data < sum_label_data
 
 
-def test_connected_label_regions():
-    shape = (13, 11, 12)
-    affine = np.eye(4)
-    n_regions = 9
-    labels_img = generate_labeled_regions(
-        shape, affine=affine, n_regions=n_regions
-    )
+def test_connected_label_regions(labels_img):
```

Comment:
Split in two also: `test_connected_label_regions_no_min_size` and `test_connected_label_regions_min_size`

Reply:
I would not as the last assert compares the differences between the 2 ways of extracting regions.

Will make the variable names even more explicit to make it harder to miss.

```python
assert n_labels_wo_min > n_labels_with_min
```